apps/common/models/file-upload.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const url = require('url');
+
+const Model = require('hmpo-model');
+const config = require('../../../config');
+
+module.exports = class UploadModel extends Model {
+  save() {
+    return new Promise((resolve, reject) => {
+      const attributes = {
+        url: config.upload.hostname
+      };
+      const reqConf = url.parse(this.url(attributes));
+      reqConf.formData = {
+        upload: {
+          value: this.get('data'),
+          options: {
+            filename: this.get('name'),
+            contentType: this.get('mimetype')
+          }
+        }
+      };
+      reqConf.method = 'POST';
+      this.request(reqConf, (err, data) => {
+        if (err) {
+          reject(err);
```

Comment:
It doesn't strictly matter because promises can only complete once, but I'll change it anyway for clarity.